plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/SiriusDiagramByRepresentationAndEObjectProvider.java

```diff
@@ -0,0 +1,203 @@
+	 */
+	public static String sanitizeFilename(String name) {
+		return name.replaceAll("[:\\\\/*?|<>]", "_");
+	}
+
+	/**
+	 * Get the full path (starting from the root folder) of the generated image
+	 * for a diagram
+	 * 
+	 * @param diagram
+	 * @param rootPath the project root path from which document generation has been launched.
+	 * @return Full path starting from root folder
+	 */
+	private String getDiagramImageFilename(DSemanticDiagram diagram, String rootPath) {
+		return rootPath + "/generated/images/representations/diagram_"
+				+ sanitizeFilename(diagram.eResource().getURIFragment(diagram)) + "." + ImageFileFormat.JPEG.getName();
+	}
+
+	/**
+	 * Retrieve all representations whose target is the specified EObject
+	 * 
+	 * @param object
+	 *            Object which is the target of the representations
+	 * @param session the Sirius session from which we want to find the representation with the given name.
+	 * @return List of representations
+	 */
+	private List<DRepresentation> getAssociatedRepresentationByName(EObject targetRootObject, String diagramDescriptionName, Session session) {
+		if (diagramDescriptionName == null) {
+			return new ArrayList<DRepresentation>(0);
+		}
+		List<DRepresentation> representationsResult = new ArrayList<DRepresentation>();
+		Collection<DRepresentation> representations = DialectManager.INSTANCE.getRepresentations(targetRootObject, session);
+
+		// Filter representations to keep only those in a selected viewpoint
+		Collection<Viewpoint> selectedViewpoints = session.getSelectedViewpoints(false);
+
+		for (DRepresentation representation : representations) {
+			if (representation instanceof DDiagram &&diagramDescriptionName.equals(((DDiagram)representation).getDescription().getName()) && representation.eContainer() instanceof DView) {
+				DView dView = (DView) representation.eContainer();
+				Viewpoint vp = dView.getViewpoint();
+				if (selectedViewpoints.contains(vp)) {
+					representationsResult.add(representation);
+				}
+			}
+		}
+		return representationsResult;
+	}
+
+	/**
+	 * Get the GmF Diagram instance corresponding to a viewpoint
+	 * DSemanticDiagram
+	 * 
+	 * @param semanticDiagram
+	 * @return
+	 */
+	private Diagram getGmfDiagram(DSemanticDiagram semanticDiagram) {
+		for (final AnnotationEntry annotation : new DDiagramQuery(semanticDiagram)
+				.getAnnotation(CustomDataConstants.GMF_DIAGRAMS)) {
+			EObject eObject = annotation.getData();
+			if (eObject instanceof Diagram) {
+				final Diagram diagramInResource = (Diagram) eObject;
+				final EObject semanticElement = ViewUtil.resolveSemanticElement(diagramInResource);
+				if (semanticElement.equals(semanticDiagram)) {
+					return diagramInResource;
+				}
+			}
+		}
+		return null;
+	}
+
+	@Override
+	public List<String> getRepresentationImagePath(Map<String, Object> parameters) throws ProviderException {
+
+		EObject rootObject = (EObject) parameters.get(ProviderConstants.KEY_CONF_ROOT_OBJECT);
```

Comment:
In this provider specifically, we shouldn't use the rootObject. The targetRootObject can be used to get the session. Using both make things unclear.